File: BookStore-Angular-MongoDb/angular/src/app/book/models/create-update-book-dto.ts
import { BookType } from './book-type';
export class CreateUpdateBookDto  {
  name: string;
  type: any;
  publishDate: string;
  price: number;

  constructor(initialValues: Partial<CreateUpdateBookDto> = {}) {
    if (initialValues) {
      for (const key in initialValues) {
        if (initialValues.hasOwnProperty(key)) {
          this[key] = initialValues[key];
        }
      }
    }
  }
}

File: BookStore-Blazor-EfCore/src/Acme.BookStore.Application.Contracts/Books/CreateUpdateBookDto.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace Acme.BookStore.Books;

[Serializable]
public class CreateUpdateBookDto
{
    [Required]
    [StringLength(128)]
    public string Name { get; set; }

    [Required]
    public BookType Type { get; set; } = BookType.Undefined;

    [Required]
    [DataType(DataType.Date)]
    public DateTime PublishDate { get; set; } = DateTime.Now;

    [Required]
    public float Price { get; set; }

    public Guid AuthorId { get; set; }
}
